torchgeo/datasets/flair2.py

```diff
@@ -0,0 +1,762 @@
+            'sentinels': 'flair_2_toy_dataset/flair_2_toy_sen_train',
+            'masks': 'flair_2_toy_dataset/flair_2_toy_labels_train',
+        },
+        'test': {
+            'images': 'flair_2_toy_dataset/flair_2_toy_aerial_test',
+            'sentinels': 'flair_2_toy_dataset/flair_2_toy_sen_test',
+            'masks': 'flair_2_toy_dataset/flair_2_toy_labels_test',
+        },
+    }
+    centroids_file: str = 'flair_2_toy_dataset/flair-2_centroids_sp_to_patch'
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        bands: Sequence[str] = FLAIR2.all_bands,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+        use_sentinel: bool = False,
+    ) -> None:
+        """Initialize a new FLAIR2Toy dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            split: which split to load, one of 'train' or 'test'
+            bands: which bands to load (B01, B02, B03, B04, B05)
+            transforms: optional transforms to apply to sample
+            download: whether to download the dataset if it is not found
+            checksum: whether to verify the dataset using checksums
+            use_sentinel: whether to use sentinel data in the dataset # FIXME: sentinel does not work with dataloader due to varying dimensions
+
+        Raises:
+            DatasetNotFoundError
+
+        ..versionadded:: 0.7
+        """
+        print('-' * 80)
+        print('WARNING: Using toy dataset.')
+        print('This dataset should be used for testing purposes only.')
+        print(
+            'Disabling use_toy-flag when initializing the dataset will initialize the full dataset.'
+        )
+        print('-' * 80)
+        super().__init__(
+            root, split, bands, transforms, download, checksum, use_sentinel
+        )
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        if os.path.isdir(os.path.join(self.root, 'flair_2_toy_dataset')):
+            print(os.path.join(self.root, 'flair_2_toy_dataset'))
+            print('Toy dataset downloaded and extracted already...')
+            return
+
+        if os.path.isfile(os.path.join(self.root, 'flair_2_toy_dataset.zip')):
+            print('Extracting toy dataset...')
+            self._extract()
+            return
+
+        if not self.download:
+            raise DatasetNotFoundError(self)
+
+        self._download('flair_2_toy_dataset')
+        self._extract()
+
+    def _download(self, url: str, suffix: str = '.zip') -> None:
+        """Download the dataset."""
+        download_url(
+            os.path.join(self.url_prefix, f'{url}{suffix}'),
+            self.root,
+            md5=self.md5s.get(url, None) if self.checksum else None,
+        )
+        # FIXME: Why is download_url not checking integrity (tests run through)?
```

Comment:
how about this FIXME?

Reply:
I left this here as somewhat of a reminder, because it has the same behavior as the parent class. I.e. currently all tests pass even with wrong md5s.